src/libs/Formula.ts

```diff
@@ -0,0 +1,458 @@
+ * Based on OldDot Formula.parsePart method
+ */
+function parsePart(definition: string): FormulaPart {
+    const part: FormulaPart = {
+        definition,
+        type: FORMULA_PART_TYPES.FREETEXT,
+        fieldPath: [],
+        functions: [],
+    };
+
+    // If it doesn't start and end with braces, it's free text
+    if (!definition.startsWith('{') || !definition.endsWith('}')) {
+        return part;
+    }
+
+    // Remove the braces and trim
+    const cleanDefinition = definition.slice(1, -1).trim();
+    if (!cleanDefinition) {
+        return part;
+    }
+
+    // Split on | to separate functions
+    const segments = cleanDefinition.split('|');
+    const fieldSegment = segments[0];
+    const functions = segments.slice(1);
+
+    // Split the field segment on : to get the field path
+    const fieldPath = fieldSegment.split(':');
+    const type = fieldPath[0]?.toLowerCase();
+
+    // Determine the formula part type
+    if (type === 'report') {
+        part.type = FORMULA_PART_TYPES.REPORT;
+    } else if (type === 'field') {
+        part.type = FORMULA_PART_TYPES.FIELD;
+    } else if (type === 'user') {
+        part.type = FORMULA_PART_TYPES.USER;
+    }
+
+    // Set field path (excluding the type)
+    part.fieldPath = fieldPath.slice(1);
+    part.functions = functions;
+
+    return part;
+}
+
+/**
+ * Compute the value of a formula given a context
+ */
+function compute(formula: string, context: FormulaContext): string {
+    if (!formula || typeof formula !== 'string') {
+        return '';
+    }
+
+    const parts = parse(formula);
+    let result = '';
+
+    for (const part of parts) {
+        let value = '';
+
+        switch (part.type) {
+            case FORMULA_PART_TYPES.REPORT:
+                value = computeReportPart(part, context);
+                break;
+            case FORMULA_PART_TYPES.FIELD:
+                value = computeFieldPart(part, context);
+                break;
+            case FORMULA_PART_TYPES.USER:
+                value = computeUserPart(part, context);
+                break;
+            case FORMULA_PART_TYPES.FREETEXT:
+                value = part.definition;
+                break;
+            default:
+                // If we don't recognize the part type, use the original definition
+                value = part.definition;
+        }
+
+        // Apply any functions to the computed value
+        value = applyFunctions(value, part.functions);
+        result += value;
+    }
+
+    return result;
+}
+
+/**
+ * Compute the value of a report formula part
+ */
+function computeReportPart(part: FormulaPart, context: FormulaContext): string {
+    const {report, policy} = context;
+    const [field, format] = part.fieldPath;
+
+    if (!field) {
+        return part.definition;
+    }
+
+    switch (field.toLowerCase()) {
+        case 'type':
+            return 'Expense Report'; // Default report type for now
+        case 'startdate':
```

Comment:
Eventually we need to support enddate as well, but we can do that in a follow up to keep this more modular.